### PR TITLE
Prepare npm publishing

### PR DESCRIPTION
**Pull request by stonko1994**
## Description
This PR will add required changes to enable npm publishing.

### CODE REVIEW

**Review by protyposis — COMMENTED** [review]
There are a few odd things with the `package.json` here:
 * This cannot be used as local package because the `main` and `types` entries point to files that do not exist, i.e. they only exist after the file has been copied to `dist`
 * The other way round, the `scripts` entries do not make sense within the `dist` folder and npm package
 * Why does `main` not point to `bitmovinplayer-analytics-conviva.js`? Why is there no `bitmovinplayer-analytics-conviva.d.ts`?

**Comment by stonko1994** [comment]
Re: 
> This cannot be used as local package because the main and types entries point to files that do not exist, i.e. they only exist after the file has been copied to dist

Where should they point then? As I understood it they should point to the `dist` (after transpiling) files. This is also the same case in the UI package

---

Re:
> The other way round, the scripts entries do not make sense within the dist folder and npm package

True

---

Re: 
> Why does main not point to bitmovinplayer-analytics-conviva.js? Why is there no bitmovinplayer-analytics-conviva.d.ts?

Why do we need the `bitmovinplayer-analytics-conviva.d.ts`? There are all types in the files included in `d.ts` files in the `dist` folder.

> Why does main not point to bitmovinplayer-analytics-conviva.js

Should it point to that? I'm not sure if there is a difference? (also taken from the UI)

**Comment by protyposis** [comment]
> Where should they point then? As I understood it they should point to the dist (after transpiling) files. This is also the same case in the UI package

They should point to files that exist after running the build task.

> Why do we need the `bitmovinplayer-analytics-conviva.d.ts`? There are all types in the files included in `d.ts` files in the `dist` folder.

Because TS does automatically load types from `d.ts` files that live along `js` files. The file can simply reexport from `index.d.ts`.

> Should it point to that? I'm not sure if there is a difference? (also taken from the UI)

Yes the compiled production build should be the default. Everybody is still free to import single files explicitly from `lib` if there is any reason for that. (The UI isn't perfect either ;) )

**Review by protyposis — CHANGES_REQUESTED** [review]
As outlined above.

**Review by reevn — COMMENTED** [review]
I'm just a silent watcher because this stuff interests me and I want to learn more about it. Don't wait for any real review from me 🙂

File: webpack.common.js
Comment (protyposis) [inline webpack.common.js]: Trailing `;` missing

**Review by protyposis — APPROVED** [review]
Looking good now! (just a minor cosmetic issue)

**Review by unki2aut — APPROVED** [review]
LGTM :+1:

**Pull request merged**
